homeassistant/components/media_player/cast.py

```diff
@@ -151,40 +150,49 @@ def _setup_internal_discovery(hass: HomeAssistantType) -> None:
 
     import pychromecast
 
-    def internal_callback(name):
-        """Handle zeroconf discovery of a new chromecast."""
-        mdns = listener.services[name]
-        _discover_chromecast(hass, ChromecastInfo(
-            host=mdns[0],
-            port=mdns[1],
-            uuid=mdns[2],
-            model_name=mdns[3],
-            friendly_name=mdns[4],
-        ))
+    browser = None
+
+    def start_discovery(event):
+        """Start discovery of new chromecasts."""
+        def internal_callback(name):
+            """Handle zeroconf discovery of a new chromecast."""
+            mdns = listener.services[name]
+            _discover_chromecast(hass, ChromecastInfo(
+                host=mdns[0],
+                port=mdns[1],
+                uuid=mdns[2],
+                model_name=mdns[3],
+                friendly_name=mdns[4],
+            ))
+
+        _LOGGER.debug("Starting internal pychromecast discovery.")
+        nonlocal browser
+        listener, browser = pychromecast.start_discovery(internal_callback)
 
-    _LOGGER.debug("Starting internal pychromecast discovery.")
-    listener, browser = pychromecast.start_discovery(internal_callback)
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_discovery)
```

Comment:
Why would we wait till start? That's generally a bad idea because that's when automations start, so any device discovered after that could trigger an automation.